packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -30,26 +23,177 @@ import type {
 	NavigatorProviderProps,
 	NavigatorLocation,
 	NavigatorContext as NavigatorContextType,
+	NavigateOptions,
 	Screen,
+	NavigateToParentOptions,
 } from '../types';
 
 type MatchedPath = ReturnType< typeof patternMatch >;
-type ScreenAction = { type: string; screen: Screen };
+type RouterAction =
```

Comment:
I assume you don't expect these types to have any utility outside of this module, which is why you opted to keep them private?